nrf-hal-common/src/pwm.rs

```diff
@@ -92,7 +93,16 @@ where
         self.pwm.countertop.read().countertop().bits()
     }
 
-    /// Sets the PWM output frequency.
+    /// Sets the PWM output period from the given input
+    /// frequency. (This calculates the period by dividing
+    /// the current PWM clock rate by `freq`.)
+    ///
+    /// This calculation takes the current [Prescaler]
+    /// setting into account.
+    ///
+    /// Subsequent use of `set_duty_on_common()`,
+    /// `set_duty_on_group()`, *etc* will override this duty
+    /// cycle setting.
```

Comment:
What duty cycle setting? Isn't this setting the frequency rather than the duty cycle?